### ISSUE TITLE
`unusedArguments` regression on `develop`

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#1495]
There seems to have been a regression in the `unusedArguments` rule on master. Here's an example that is now incorrectly marked as unused:

```swift
private func scrollViewDidEndDecelerating(_ visibleDayRange: DayRange) {
    guard
        store.state.request.isIdle,
        let nextDayToLoad = store.state.request.nextCursor?.lowerBound,
        visibleDayRange.upperBound.distance(to: nextDayToLoad) < 30
    else { return }

    store.handle(.loadNext)
}
```

Here's a new test case that fails but should pass:

```swift
func testUnusedArguments() {
    let input = """
    private func scrollViewDidEndDecelerating(_ visibleDayRange: DayRange) {
        guard
            store.state.request.isIdle,
            let nextDayToLoad = store.state.request.nextCursor?.lowerBound,
            visibleDayRange.upperBound.distance(to: nextDayToLoad) < 30
        else { return }

        store.handle(.loadNext)
    }
    """

    testFormatting(for: input, rule: FormatRules.unusedArguments)
}
```

**Comment by calda** [comment]
Here's another example that fails on `develop`:

```swift
func testUnusedArguments() {
    let input = """
    let onSuccess: ([AIRReservation]?, Bool) -> Void = { reservations, _ in
        if
            let reservations,
            eligibleToShow(
                reservations,
                accountService: accountService)
          {
            coordinator.startFlow()
        }
    }
    """

    testFormatting(for: input, rule: FormatRules.unusedArguments)
}
```

**Comment by calda** [comment]
thanks for the fix @nicklockwood!

**Comment by nicklockwood** [comment]
@calda thanks for reporting it! Fix landed in 0.51.14

**Comment by nicklockwood** [comment]
@calda due to a merge screwup on my part, the fix was not actually included in 0.51.14. It's included in 0.51.15 instead.
